parser/src/Parse/Module.hs

```diff
@@ -181,7 +181,11 @@ import' =
     exposing =
       do  preExposing <- try (whitespace <* reserved "exposing")
           postExposing <- whitespace
-          imports <- listing detailedListing
+          imports <-
+            choice
```

Comment:
Since you asked for feedback:  The thing I'm thinking about when reviewing this code is what potential performance implication this `choice` has (which depends on whether the early choices allow for significant backtracking (with `try`) or not).  In this case, `listing` does `try (char '(')` so I think that will do a maximum of 1 character of backtracking, which should be fine.  (If there were the possibility of significant backtracking, then I'd want to look at ways to rewrite the choice to avoid the backtracking.)

In any case, I think it looks good for this particular change.

Reply:
Great, thanks for the info. I also thought about the order, and having the standard case at first, failing at the missing "(".